domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/eventcasefieldcomplextype/EventComplexTypeEntityDisplayContextParameterValidatorImpl.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.eventcasefieldcomplextype;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter.DisplayContextParameter;
+import uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter.DisplayContextParameterType;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.displaycontextparameter.AbstractDisplayContextParameterValidator;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.displaycontextparameter.DisplayContextParameterValidatorFactory;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.eventcasefield.EventCaseFieldEntityValidationContext;
+import uk.gov.hmcts.ccd.definition.store.repository.DisplayContext;
+import uk.gov.hmcts.ccd.definition.store.repository.FieldTypeUtils;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.ComplexFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.EventComplexTypeEntity;
+
+import java.util.Optional;
+
+@Component
+public class EventComplexTypeEntityDisplayContextParameterValidatorImpl extends AbstractDisplayContextParameterValidator<EventComplexTypeEntity> implements EventComplexTypeEntityValidator {
```

Comment:
did we leave `DateTime` intentionally out for brevity?

Reply:
This class could be used in future for any other DCP with little change so was kept out on purpose. (With the scope change this specific class has since become the template of the new `ComplexFieldEntityDisplayContextParameterValidatorImpl`)